File: src/lib/tauri-api/coordinator.ts
import type {
  ReplayWorkspaceCoordinatorActionInput,
  StartWorkspaceCoordinatorInput,
  StepWorkspaceCoordinatorInput,
  WorkspaceCoordinatorStatus,
} from '../../types/coordinator';
import { invokeCommand } from './client';

export function getWorkspaceCoordinatorStatus(workspaceId: string): Promise<WorkspaceCoordinatorStatus> {
  return invokeCommand<WorkspaceCoordinatorStatus>('get_workspace_coordinator_status', { workspaceId });
}

export function startWorkspaceCoordinator(input: StartWorkspaceCoordinatorInput): Promise<WorkspaceCoordinatorStatus> {
  return invokeCommand<WorkspaceCoordinatorStatus>('start_workspace_coordinator', { input });
}

export function stepWorkspaceCoordinator(input: StepWorkspaceCoordinatorInput): Promise<WorkspaceCoordinatorStatus> {
  return invokeCommand<WorkspaceCoordinatorStatus>('step_workspace_coordinator', { input });
}

export function stopWorkspaceCoordinator(workspaceId: string): Promise<WorkspaceCoordinatorStatus> {
  return invokeCommand<WorkspaceCoordinatorStatus>('stop_workspace_coordinator', { workspaceId });
}

export function replayWorkspaceCoordinatorAction(
  input: ReplayWorkspaceCoordinatorActionInput,
): Promise<WorkspaceCoordinatorStatus> {
  return invokeCommand<WorkspaceCoordinatorStatus>('replay_workspace_coordinator_action', { input });
}

File: src-tauri/src/commands/coordinator.rs
use tauri::State;

use crate::models::{
    ReplayWorkspaceCoordinatorActionInput, StartWorkspaceCoordinatorInput,
    StepWorkspaceCoordinatorInput, WorkspaceCoordinatorStatus,
};
use crate::services::coordinator_service;
use crate::state::AppState;

#[tauri::command]
pub fn get_workspace_coordinator_status(
    state: State<'_, AppState>,
    workspace_id: String,
) -> Result<WorkspaceCoordinatorStatus, String> {
    coordinator_service::get_workspace_coordinator_status(&state, &workspace_id)
}

#[tauri::command]
pub fn start_workspace_coordinator(
    state: State<'_, AppState>,
    input: StartWorkspaceCoordinatorInput,
) -> Result<WorkspaceCoordinatorStatus, String> {
    coordinator_service::start_workspace_coordinator(&state, input)
}

#[tauri::command]
pub fn step_workspace_coordinator(
    state: State<'_, AppState>,
    input: StepWorkspaceCoordinatorInput,
) -> Result<WorkspaceCoordinatorStatus, String> {
    coordinator_service::step_workspace_coordinator(&state, input)
}

#[tauri::command]
pub fn stop_workspace_coordinator(
    state: State<'_, AppState>,
    workspace_id: String,
) -> Result<WorkspaceCoordinatorStatus, String> {
    coordinator_service::stop_workspace_coordinator(&state, &workspace_id)
}

#[tauri::command]
pub fn replay_workspace_coordinator_action(
    state: State<'_, AppState>,
    input: ReplayWorkspaceCoordinatorActionInput,
) -> Result<WorkspaceCoordinatorStatus, String> {
    coordinator_service::replay_workspace_coordinator_action(&state, input)
}
